qa/rpc-tests/p2p_nu_peer_management.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python2
+# Copyright (c) 2018 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.mininode import (
+    NodeConn,
+    NodeConnCB,
+    NetworkThread,
+    msg_ping,
+    SPROUT_PROTO_VERSION,
+    OVERWINTER_PROTO_VERSION,
+    SAPLING_PROTO_VERSION,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import initialize_chain_clean, start_nodes, \
+    p2p_port, assert_equal
+
+import time
+
+#
+# In this test we connect Sprout, Overwinter, and Sapling mininodes to a Zcashd
+# node which will activate Overwinter at block 10 and Sapling at block 15.
+#
+# We test:
+# 1. the mininodes stay connected to Zcash with Sprout consensus rules
+# 2. when Overwinter activates, the Sprout mininodes are dropped
+# 3. new Overwinter and Sapling nodes can connect to Zcash
+# 4. new Sprout nodes cannot connect to Zcash
+# 5. when Sapling activates, the Overwinter mininodes are dropped
+# 6. new Sapling nodes can connect to Zcash
+# 7. new Sprout and Overwinter nodes cannot connect to Zcash
+#
+# This test *does not* verify that prior to each activation, the Zcashd
+# node will prefer connections with NU-aware nodes, with an eviction process
+# that prioritizes non-NU-aware connections.
+#
+
+
+class TestManager(NodeConnCB):
+    def __init__(self):
+        NodeConnCB.__init__(self)
+        self.create_callback_map()
+
+    def on_close(self, conn):
+        pass
+
+    def on_reject(self, conn, message):
+        conn.rejectMessage = message
+
+
+class NUPeerManagementTest(BitcoinTestFramework):
```

Comment:
Nit: It's easier to review if the move and the changes are separate commits.

Reply:
Sorry, I forgot that GitHub doesn't detect and render it as a moved file if there are any changes.